compiler/passes/src/code_generation/visit_expressions.rs

```diff
@@ -490,32 +512,34 @@ impl<'a> CodeGenerator<'a> {
     }
 
     fn visit_call(&mut self, input: &'a CallExpression) -> (String, String) {
-        // Need to determine the program the function originated from as well as if the function has a finalize block.
-        let (mut call_instruction, has_finalize);
+        // Lookup the function return type.
```

Comment:
Should `CallExpression` use `Location` now?

Reply:
Is it wasteful to store the name of the function twice?

Reply:
Since `function.function` needs to capture the span and node_id but location just captures the Symbols representing program name and function name

Reply:
I will refactor `function.function` to be an identifier instead of an expression though.

Reply:
Actually maybe its better to keep it as an expression because otherwise the parser will need to panic in some cases (when non-identifier function is supplied) when we would prefer it keep spitting out errors.

Reply:
Ah good point, should it be `LocatorExpression`?